improver_tests/test_field_texture.py

```diff
@@ -0,0 +1,244 @@
+
+    with pytest.raises(Exception) as excinfo:
+        PLUGIN.process(non_binary_cube)
+    assert str(excinfo.value) == "Incorrect input. Cube should hold binary data only"
+
+
+def test_process_no_cloud(no_cloud_cube):
+    """Test the FieldTexture plugin with multi realization input cube that has
+       no cloud present in the field."""
+
+    expected_data = np.ones(no_cloud_cube[0][0].shape)
+
+    result = PLUGIN.process(no_cloud_cube)
+    np.testing.assert_almost_equal(result.data, expected_data, decimal=4)
+
+
+def test_process_all_cloud(all_cloud_cube):
+    """Test the process function with multi realization input cube that has
+       all cloud occupying the field."""
+
+    expected_data = np.zeros(all_cloud_cube[0][0].shape)
+
+    result = PLUGIN.process(all_cloud_cube)
+    np.testing.assert_almost_equal(result.data, expected_data, decimal=4)
+
+
+def test_no_threshold_cube(multi_cloud_cube):
+    """Test the process function with multi_cloud_cube that has no thresholds"""
+
+    multi_cloud_cube.remove_coord("cloud_area_fraction")
+    with pytest.raises(CoordinateNotFoundError, match="No threshold coord found.*"):
+        PLUGIN.process(multi_cloud_cube)
+
+
+def test_wrong_threshold(multi_cloud_cube):
+    """Test the process function with multi_cloud_cube where user defined
+        diagnostic_threshold variable does not match threshold available in
+        the cube."""
+
+    plugin = FieldTexture(
+        nbhood_radius=NB_RADIUS,
+        textural_threshold=TEXT_THRESH,
+        diagnostic_threshold="0.235",
+    )
+    with pytest.raises(ValueError, match="Threshold 0.235 is not present.*"):
+        plugin.process(multi_cloud_cube)
+
+
+def test_metadata_name(multi_cloud_cube):
+    """Test that the metadata of the output cube follows expected conventions
+       after the plugin is complete and all old coordinates have been removed."""
+
+    expected_name = "probability_of_texture_of_cloud_area_fraction_above_threshold"
+    expected_units = "1"
+
+    result = PLUGIN.process(multi_cloud_cube)
+
+    # check name and units
+    result_name = result.name()
+    assert expected_name == result_name
+    result_units = result.units
+    assert expected_units == result_units
+
+
+def test_metadata_coords(multi_cloud_cube):
+    """Test that the coordinate metadata in the output cube follows expected
+        conventions after that plugin has completed."""
+
+    expected_threshold_coord_name = "texture_of_cloud_area_fraction"
+    expected_units = "1"
+
+    result = PLUGIN.process(multi_cloud_cube)
+
+    # check coordinates
+    assert "texture_of_cloud_area_fraction" in [
+        crd.name() for crd in result.coords(dim_coords=False)
+    ]
+    coord_names = [crd.name() for crd in result.coords()]
+    assert expected_threshold_coord_name in coord_names
+    for coord in ["cloud_area_fraction", "realization"]:
+        assert coord not in coord_names
```

Comment:
I am tempted to be more complete here, ie:
```
expected_dims = ["projection_y_coordinate", "projection_x_coordinate"]
expected_scalar_coords = ["texture_of_cloud_area_fraction"]
result_dims = [coord.name() for coord in result.coords(dim_coords=True)]
result_scalar_coords = [coord.name() for coord in result.coords(dim_coords=False)]
assert result_dims == expected_dims
assert result_scalar_coords == expected_scalar_coords
```
Or whatever assert statement you need to compare lists.

Reply:
You can then get rid of `expected_threshold_coord_name = ...` above.

Reply:
Done.